opentelekomcloud/services/elb/v3/data_source_opentelekomcloud_lb_certificate_v3.go

```diff
@@ -0,0 +1,139 @@
+package v3
+
+import (
+	"context"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/opentelekomcloud/gophertelekomcloud/openstack/elb/v3/certificates"
+	"github.com/opentelekomcloud/terraform-provider-opentelekomcloud/opentelekomcloud/common"
+	"github.com/opentelekomcloud/terraform-provider-opentelekomcloud/opentelekomcloud/common/cfg"
+	"github.com/opentelekomcloud/terraform-provider-opentelekomcloud/opentelekomcloud/common/fmterr"
+)
+
+func DataSourceCertificateV3() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: dataSourceCertificateV3Read,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringLenBetween(0, 255),
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"domain": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"private_key": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Computed:         true,
+				DiffSuppressFunc: common.SuppressStrippedNewLines,
+			},
+			"certificate": {
+				Type:             schema.TypeString,
+				Optional:         true,
+				Computed:         true,
+				DiffSuppressFunc: common.SuppressStrippedNewLines,
+			},
+			"type": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"server", "client",
+				}, false),
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"updated_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"expire_time": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceCertificateV3Read(_ context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	config := meta.(*cfg.Config)
+	client, err := config.ElbV3Client(config.GetRegion(d))
+	if err != nil {
+		return fmterr.Errorf(ErrCreateClient, err)
+	}
+
+	id := d.Get("id").(string)
+	if id != "" {
+		cert, err := certificates.Get(client, id).Extract()
+		if err != nil {
+			return fmterr.Errorf("error getting ELBv3 certificate: %w", err)
+		}
+		return setCertificateFields(d, cert)
+	}
+
+	listOpts := certificates.ListOpts{
+		Name:        d.Get("name").(string),
+		Type:        d.Get("type").(string),
+		Domain:      d.Get("domain").(string),
+		PrivateKey:  d.Get("private_key").(string),
+		Certificate: d.Get("certificate").(string),
+	}
```

Comment:
`ListOpts` seems to be different to the docs:
https://docs.otc.t-systems.com/api/elb/ListCertificates.html